lib/wallet/path.js

```diff
@@ -212,14 +213,23 @@ class Path extends bio.Struct {
 
   /**
    * Convert path object to string derivation path.
+   * @param {String|Network?} network - Network type.
    * @returns {String}
    */
 
-  toPath() {
+  toPath(network) {
     if (this.keyType !== Path.types.HD)
       return null;
 
-    return `m/${this.account}'/${this.branch}/${this.index}`;
+    let prefix = 'm';
+
+    if (network) {
+      const purpose = 44;
+      network = Network.get(network);
+      prefix += `/${purpose}'/${network.keyPrefix.coinType}'`;
+    }
+
+    return `${prefix}/${this.account}'/${this.branch}/${this.index}`;
```

Comment:
This might be an issue whenever `this.account != this.accountKey.accountDepth`..